.github/workflows/python-publish.yml

```diff
@@ -0,0 +1,42 @@
+# This workflow will upload a Python Package using Twine when a release is created
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python#publishing-to-package-registries
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Upload Python Package
+
+on:
+  release:
+    types: [published]
+
+permissions:
+  contents: read
+
+jobs:
+  deploy:
+
+    runs-on: ubuntu-latest
```

Comment:
We will need support from Linux, MacOS, and Windows systems.

Reply:
Do we need to test on multiple versions of each OS or testing on the latest version of each OS will be sufficient?

Reply:
Testing on the latest version for each OS would be sufficient.

Reply:
While uploading to pypi do we need to upload wheels of all possible combinations of python versions with OS, or just 1 Source distribution(tar.gz) and 1 Built distribution (.whl)?
And, which python versions are we testing on?

Reply:
@FanwangM if we want wheels for all the 3 OS, then we would have to change the `setup.py` file to output the wheels with dynamic naming

Reply:
Can you please help change `setup.py`? Thanks. @kunikachandra

Reply:
Yes, sure! I will start working on it once my exams are over in 2-3 days